sn_networking/src/record_store.rs

```diff
@@ -428,27 +428,41 @@ impl NodeRecordStore {
     /// Calculate the cost to store data for our current store state
     #[allow(clippy::mutable_key_type)]
     pub(crate) fn store_cost(&self, key: &Key) -> (NanoTokens, QuotingMetrics) {
-        let quoting_metrics = QuotingMetrics {
-            records_stored: self.records.len(),
-            max_records: self.config.max_records,
-            received_payment_count: self.received_payment_count,
-            live_time: self.timestamp.elapsed().as_secs(),
-        };
+        let live_time = self.timestamp.elapsed().as_secs();
+        let mut records_stored = self.config.max_records;
 
         let cost = if self.contains(key) {
-            0
+            NanoTokens::zero()
+        } else if let Some(distance_range) = self.distance_range {
+            let record_keys_as_hashset: HashSet<&Key> = self.records.keys().collect();
+            let relevant_records =
```

Comment:
note that there is `pruning` to be carried out when hit max_records, which is regarding ALL.
i.e. this means `relevant_records` may way below max_records, and will never reach that bar.

It may be good, may be bad,

Also, I'd suggest just rename `QuotingMetrics::records_stored` to `QuotingMetrics::relevant_records` to avoid any potential confusion.